xcdat/axis.py

```diff
@@ -180,68 +226,151 @@ def swap_lon_axis(
         The Dataset with swapped lon axes orientation.
     """
     ds = dataset.copy()
-    lon: xr.DataArray = get_axis_coord(ds, "X").copy()
-    lon_bounds: xr.DataArray = dataset.bounds.get_bounds("X").copy()
+    coord_keys = get_axis_coord(ds, "X").coords.keys()
 
+    # Attempt to swap the orientation for longitude coordinates.
     with xr.set_options(keep_attrs=True):
-        if to == (-180, 180):
-            new_lon = ((lon + 180) % 360) - 180
-            new_lon_bounds = ((lon_bounds + 180) % 360) - 180
-            ds = _reassign_lon(ds, new_lon, new_lon_bounds)
-        elif to == (0, 360):
-            new_lon = lon % 360
-            new_lon_bounds = lon_bounds % 360
-            ds = _reassign_lon(ds, new_lon, new_lon_bounds)
-
-            # Handle cases where a prime meridian cell exists, which can occur
-            # after swapping to (0, 360).
-            p_meridian_index = _get_prime_meridian_index(new_lon_bounds)
-            if p_meridian_index is not None:
-                ds = _align_lon_to_360(ds, p_meridian_index)
-        else:
-            raise ValueError(
-                "Currently, only (-180, 180) and (0, 360) are supported longitude axis "
-                "orientations."
-            )
+        for key in coord_keys:
+            new_coord = _swap_lon_axis(ds.coords[key], to)
+
+            if ds.coords[key].identical(new_coord):
+                continue
 
-    # If the swapped axis orientation is the same as the existing axis
-    # orientation, return the original Dataset.
-    if new_lon.identical(lon):
-        return dataset
+            ds.coords[key] = new_coord
+
+        # Attempt to swap the orientation for longitude coordinate bounds.
+        # NOTE: This logic assumes each dimension coordinate is mapped to one
+        # bounds key.
+        try:
+            bounds = ds.bounds.get_bounds("X")
+            bounds_key = bounds.name
+
+            new_bounds = _swap_lon_axis(bounds, to)
+
+            # FIXME: Attempting to swap (0, 360) to (0, 360) should be skipped.
+            # However, `_swap_lon_axis` uses coord % 360 to convert to 360 axis.
+            # If there are cells with 360 values, it will be converted to 0,
+            # which means this conditional fails.
+            if not ds[bounds_key].identical(new_bounds):
+                ds[bounds_key] = new_bounds
+                # Handle cases where a prime meridian cell exists, which can occur
+                # after swapping longitude bounds to (0, 360).
+                if to == (0, 360):
+                    p_meridian_index = _get_prime_meridian_index(ds[bounds_key])
+
+                    if p_meridian_index is not None:
+                        ds = _align_lon_to_360(ds, ds[bounds_key], p_meridian_index)
```

Comment:
Hey @pochedls, I am tackling [Phase 3 - Update Swapping Longitude Axis Orientation and Centering Time Coordinates](https://docs.google.com/document/d/1BnwNGzb-_pSUFzH5513x99Yv2IOanZlTtA7RvIF7hJA/edit#bookmark=id.wbamqmhh4jwl).

I am running into an issue where swapping coordinates that are (0, 360) to (0, 360) results in non-identical coordinates when they should be identical.

Example:
```python
# Bounds are (0, 360)
bounds

<xarray.DataArray 'lon_bnds' (lon: 3, bnds: 2)>
array([[  0, 120],
       [120, 181],
       [181, 360]])
Coordinates:
  * lon      (lon) int64 60 150 271
Dimensions without coordinates: bnds
Attributes:
    xcdat_bounds:  True

# `new_bounds = bounds % 360`, resulting in the 360 point being swapped to 0.
new_bounds

<xarray.DataArray 'lon_bnds' (lon: 3, bnds: 2)>
array([[  0, 120],
       [120, 181],
       [181,   0]])
Coordinates:
  * lon      (lon) int64 60 150 271
Dimensions without coordinates: bnds
Attributes:
    xcdat_bounds:  True

# As a result, `bounds` and `new_bounds` are non-identical.
bounds.identical(new_bounds)
False
```

**Can you think of a solution for checking whether the original axis orientation is (0, 360) or (-180, 180) without needing to compare the original coordinates to the swapped coordinates?** (e.g., do we check if there is a point close to 360?)

Reply:
You probably already looked at this, but we had to do something similar in the spatial averager ([here](https://github.com/xCDAT/xcdat/blob/main/xcdat/spatial.py#L439)). 

I don't understand the full context here, but I think that there is a special case that could be define. 

IF you are swapping to (0, 360) AND 360 is an element in the bounds, THEN after swapping to (0, 360), you can/should replace that specific element with 360.

Would that updated logic solve this issue?

Reply:
Yes that makes sense to me. 

Example logic:

1. Save the index of the element that is 360 (if sorted, this should be the last element)
2. Swap the bounds (360 becomes 0)
3. Check if bounds are circular
4. If circular, use the saved index to replace 0 with 360.

Reply:
Does it matter if things are circular (I assume you mean spanning 0 to 360 or -180 to 180)? I would think that even if you had a partial domain, e.g., `bounds = [[270, 300], [300, 330], [330, 360]]` you would still want to replace the 360 element after swapping to a 360 degree axis. Otherwise I think you'd have `bounds = [[270, 300], [300, 330], [330, 0]]`.

Or maybe you mean something else by circular.

Reply:
Simpler logic:
```python
        # Swap the coordinates.
        # Example with 180 coords: [-180, -0, 179] -> [0, 180, 360]
        # Example with 360 coords: [60, 150, 360] -> [60, 150, 0]
        new_coords = coords % 360

        # Check if the original coordinates contain an element with a value of
        # 360. If this element exists, use its index to revert its swapped
        # value of 0 (360 % 360 is 0) back to 360. This case usually happens
        # if the coordinate are already on the (0, 360) axis orientation.
        # Example with 360 coords: [60, 150, 0] -> [60, 150, 360]
        circular_index = np.where(coords == 360)
        if len(circular_index) > 0:
            new_coords[circular_index] = 360
```

Reply:
For some reason your commented didn't appear when I commented with the simpler logic.

I think my simpler logic addresses your comment though (I should rename `circular_index` to `index_with_360`).